apm-agent-plugins/apm-log-correlation-plugin/src/test/java/co/elastic/apm/agent/mdc/MdcActivationListenerIT.java

```diff
@@ -0,0 +1,172 @@
+        transaction.activate();
+        mockedLogger.error("Some slf4j exception", new RuntimeException("Hello exception"));
+
+        assertMdcErrorIdIsEmpty();
+
+        transaction.deactivate().end();
+
+        assertMdcErrorIdIsEmpty();
+    }
+
+    @Test
+    void testVerifyThatWithEnabledCorrelationAndLoggedErrorMdcErrorIdIsNotBlankWithSlf4jNotInTransaction() {
+        when(loggingConfiguration.isLogCorrelationEnabled()).thenReturn(true);
+        Logger mockedLogger = mock(Logger.class);
+        doAnswer(invocation -> assertMdcErrorIdIsNotEmpty()).when(mockedLogger).error(anyString(), any(Exception.class));
+
+        assertMdcErrorIdIsEmpty();
+
+        mockedLogger.error("Some slf4j exception", new RuntimeException("Hello exception"));
+
+        assertMdcErrorIdIsEmpty();
+    }
+
+    @Test
+    void testVerifyThatWithEnabledCorrelationAndLoggedErrorMdcErrorIdIsNotBlankWithLog4j() {
+        Assumptions.assumeTrue(() -> {
+            org.apache.log4j.MDC.put("test", true);
+            return org.apache.log4j.MDC.get("test") == Boolean.TRUE;
+        }, "Log4j MDC is not working, this happens with some versions of Java 10 where log4j thinks it's Java 1");
+        when(loggingConfiguration.isLogCorrelationEnabled()).thenReturn(true);
+        org.apache.logging.log4j.Logger logger = mock(org.apache.logging.log4j.Logger.class);
+        doAnswer(invocation -> assertMdcErrorIdIsNotEmpty()).when(logger).error(anyString(), any(Exception.class));
+
+        assertMdcErrorIdIsEmpty();
+
+        Transaction transaction = tracer.startRootTransaction(getClass().getClassLoader()).withType("request").withName("test");
+        transaction.activate();
+        logger.error("Some apache logger exception", new RuntimeException("Hello exception"));
+        assertMdcErrorIdIsEmpty();
+        transaction.deactivate().end();
+
+        assertMdcErrorIdIsEmpty();
+    }
+
+    @Test
+    void testVerifyThatWithEnabledCorrelationAndLoggedErrorMdcErrorIdIsNotBlankWithLog4jNotInTransaction() {
+        Assumptions.assumeTrue(() -> {
+            org.apache.log4j.MDC.put("test", true);
+            return org.apache.log4j.MDC.get("test") == Boolean.TRUE;
+        }, "Log4j MDC is not working, this happens with some versions of Java 10 where log4j thinks it's Java 1");
+        when(loggingConfiguration.isLogCorrelationEnabled()).thenReturn(true);
+        org.apache.logging.log4j.Logger logger = mock(org.apache.logging.log4j.Logger.class);
+        doAnswer(invocation -> assertMdcErrorIdIsNotEmpty()).when(logger).error(anyString(), any(Exception.class));
+
+        assertMdcErrorIdIsEmpty();
+
+        logger.error("Some apache logger exception", new RuntimeException("Hello exception"));
+
+        assertMdcErrorIdIsEmpty();
+    }
+
+    private void assertMdcErrorIdIsEmpty() {
+        assertThat(MDC.get("error.id")).isNull();
+        if (org.apache.log4j.MDC.get("test") == Boolean.TRUE) {
```

Comment:
This if can be removed, as we're using assumptions in the test methods now. The way assumptions work is similar to assertions but instead of failing the test, they just skip the test.

Reply:
fixed